### PR TITLE
Application Instance Move only if clustered

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

The application instances, will be moved when the host id changes AND the application instance is clustered.

This new behavior, aims to fix the 2379 bug,

## How was this tested?

Automated tests

### CODE REVIEW

File: lib/trento/application/integration/discovery/policies/sap_system_policy.ex [hunk 1]
```diff
@@ -121,6 +121,8 @@ defmodule Trento.Integration.Discovery.SapSystemPolicy do
          },
          host_id
        ) do
+    cluster_id = Trento.Clusters.get_cluster_id_by_host_id(host_id)
```
Comment (arbulu89) [inline lib/trento/application/integration/discovery/policies/sap_system_policy.ex:124]: Would it be better to run this in the `discovery.ex` file?
We do the same to get the current instances, so the policy stays 100% pure, without external side-effects

File: lib/trento/domain/sap_system/commands/register_application_instance.ex [hunk 2]
```diff
@@ -38,6 +38,7 @@ defmodule Trento.Domain.Commands.RegisterApplicationInstance do
     field :db_host, :string
     field :tenant, :string
     field :host_id, Ecto.UUID
+    field :cluster_id, Ecto.UUID
```
Comment (arbulu89) [inline lib/trento/domain/sap_system/commands/register_application_instance.ex:41]: Could we add a little doc lines in top?
Something like:

>`cluster_id` value is used to know if the application instance is clustered or not. This information is required in order to decide whether the instance was moved by the cluster in a failover scenario or not

File: test/trento/application/integration/discovery/policies/sap_system_policy_test.exs [hunk 3]
```diff
@@ -96,6 +121,30 @@ defmodule Trento.Integration.Discovery.SapSystemPolicyTest do
              |> SapSystemPolicy.handle([])
   end
 
+  test "should return the expected commands when a sap_system payload of type application and diagnostics is handled and the host is part of a cluster" do
```
Comment (arbulu89) [inline test/trento/application/integration/discovery/policies/sap_system_policy_test.exs:124]: Do we need this test?
It doesn't test any new code as far as I see

File: test/trento/domain/sap_system/sap_system_test.exs [hunk 4]
```diff
@@ -537,6 +539,91 @@ defmodule Trento.SapSystemTest do
       )
     end
 
+    test "should not move an application instance if the host_id changed and the application instance is not clustered" do
```
Comment (arbulu89) [inline test/trento/domain/sap_system/sap_system_test.exs:542]: I would redo and rephrase this test. At the end, we want to check if already registered instance numbers in new not clustered hosts are registered as new instances:

`should register a new not clustered application instance if it has an already existing instance number`

**Review by arbulu89 — COMMENTED** [review]
Some suggestions.
To have some e2e tests (other PR) we could do:
1s test:
- Register clustered ASCS. Send the same payload from other host. This should emit a movement event
- Register a not clustered APP with an already existing instance number in other host. This should register a new instance

File: lib/trento/application/integration/discovery/policies/sap_system_policy.ex (hunk 1 above)
Comment (CDimonaco) [reply lib/trento/application/integration/discovery/policies/sap_system_policy.ex]: Done

File: test/e2e/cypress/e2e/sap_systems_overview.cy.js [hunk 5]
```diff
@@ -344,4 +344,54 @@ context('SAP Systems Overview', () => {
...
+
+      cy.get('table.table-fixed > tbody > tr')
+        .eq(1)
+        .find('div.table-row-group')
+        .eq(0)
+        .find('div.table-row')
+        .its('length')
+        .should('eq', 4);
+
+        cy.contains('vmnwdev01').should('not.exist');
```
Comment (arbulu89) [inline test/e2e/cypress/e2e/sap_systems_overview.cy.js:380]: I would put this `vmnwdev01` in the `nwdSystem` const.

File: test/trento/application/integration/discovery/policies/sap_system_policy_test.exs (hunk 3 above)
Comment (arbulu89) [reply test/trento/application/integration/discovery/policies/sap_system_policy_test.exs:124]: we are still in the same. I would remove this test, as it doesn't test anything new. But well, it is not harmful neither

File: test/trento/domain/sap_system/sap_system_test.exs [hunk 6]
```diff
@@ -537,6 +539,189 @@ defmodule Trento.SapSystemTest do
...
+          tenant: tenant,
+          instance_number: instance_number,
+          instance_hostname: instance_hostname,
+          features: "ABAP",
+          http_port: http_port,
+          https_port: https_port,
+          start_priority: start_priority,
+          host_id: new_host_id,
+          health: :passing,
+          ensa_version: ensa_version
```
Comment (arbulu89) [inline test/trento/domain/sap_system/sap_system_test.exs:607]: Maybe we could put `cluster_id: nil`, to make it more obvious

**Review by arbulu89 — APPROVED** [review]
@CDimonaco Alright!
All good in my eyes.
Simple commented some few cosmetic things

File: test/trento/domain/sap_system/sap_system_test.exs (hunk 6 above)
Comment (CDimonaco) [reply test/trento/domain/sap_system/sap_system_test.exs:607]: done

File: test/e2e/cypress/e2e/sap_systems_overview.cy.js (hunk 5 above)
Comment (CDimonaco) [reply test/e2e/cypress/e2e/sap_systems_overview.cy.js:380]: done

**Review by arbulu89 — APPROVED** [review]

**Pull request merged**